packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -80,6 +81,29 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
             throw new Error("File path should be provided!!");
         }
 
+        let fileInfo: INewFileInfo | IExistingFileInfo;
+        let createShareLinkParam: ShareLinkTypes | ISharingLinkKind | undefined;
+        if (odspResolvedUrl.fileName) {
```

Comment:
I think you should first have the other condition. Because somebody can give both itemid and filename. But to create a brand new file, already existing itemid is not possible. So it should be like:
if (itemid) -> partition case
else if(filename) -> brand new file case
else error.